pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go

```diff
@@ -149,6 +149,52 @@ type TidbMonitorRef struct {
 	Name string `json:"name"`
 }
 
-// TODO: sync status
+// +k8s:openapi-gen=true
+// TidbClusterAutoSclaerStatus describe the whole status
 type TidbClusterAutoSclaerStatus struct {
+	// Tikv describes the status for the tikv in the last auto-scaling reconciliation
+	// +optional
+	TiKV *TikvAutoScalerStatus `json:"tikv,omitempty"`
+	// Tidb describes the status for the tidb in the last auto-scaling reconciliation
+	// +optional
+	TiDB *TidbAutoScalerStatus `json:"tidb,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// TidbAutoScalerStatus describe the auto-scaling status of tidb
+type TidbAutoScalerStatus struct {
+	BasicAutoScalerStatus `json:",inline"`
+}
+
+// +k8s:openapi-gen=true
+// TikvAutoScalerStatus describe the auto-scaling status of tikv
+type TikvAutoScalerStatus struct {
+	BasicAutoScalerStatus `json:",inline"`
+}
+
+// +k8s:openapi-gen=true
+// BasicAutoScalerStatus describe the basic auto-scaling status
+type BasicAutoScalerStatus struct {
+	// MetricsStatusList describes the metrics status in the last auto-scaling reconciliation
+	// +optional
+	MetricsStatusList []MetricsStatus `json:"metrics,omitempty"`
+	// CurrentReplicas describes the current replicas for the component(tidb/tikv)
+	CurrentReplicas int32 `json:"currentReplicas"`
+	// RecommendedReplicas describes the calculated replicas in the last auto-scaling reconciliation for the component(tidb/tikv)
+	// +optional
+	RecommendedReplicas *int32 `json:"recommendedReplicas,omitempty"`
+	// LastAutoScalingTimestamp describes the last auto-scaling timestamp for the component(tidb/tikv)
+	// +optional
+	LastAutoScalingTimestamp *string `json:"lastAutoScalingTimestamp,omitempty"`
```

Comment:
why use `*string` as timestamp?

Reply:
Do you mean to use `string` for timestamp or other struct?

Reply:
Can you use `*time.Time`?

Reply:
yes, then we don't need to parse it if we want to call time functions on it. by the way, it's better to use `*metav1.Time` in Kubernetes. It's a wrapper around `time.Time` which supports correct marshaling to YAML and JSON.

Reply:
good advice, updated.